flink-yarn/src/main/java/org/apache/flink/yarn/YarnClusterClient.java

```diff
@@ -211,14 +191,41 @@ public void run() {
 		Runtime.getRuntime().addShutdownHook(clientShutdownHook);
 
 		isConnected = true;
+
+		logAndSysout("Waiting until all TaskManagers have connected");
+
+		while(true) {
+			GetClusterStatusResponse status = getClusterStatus();
+			if (status != null) {
+				if (status.numRegisteredTaskManagers() < clusterDescriptor.getTaskManagerCount()) {
+					logAndSysout("TaskManager status (" + status.numRegisteredTaskManagers() + "/"
+						+ clusterDescriptor.getTaskManagerCount() + ")");
+				} else {
+					logAndSysout("All TaskManagers are connected");
+					break;
+				}
+			} else {
+				logAndSysout("No status updates from the YARN cluster received so far. Waiting ...");
+			}
+
+			try {
+				Thread.sleep(500);
+			}
+			catch (InterruptedException e) {
+				LOG.error("Interrupted while waiting for TaskManagers");
+				System.err.println("Thread is interrupted");
+				Thread.currentThread().interrupt();
```

Comment:
I guess I wrote this code. If I'm not mistaken, this means that the while(true) loop is not interruptable, right?
I think we should change that and break out of the loop if it has been interrupted.

What do you think?